app/src/main/java/org/mozilla/focus/urlinput/UrlInputPresenter.java

```diff
@@ -71,25 +74,36 @@ public void onInput(@NonNull CharSequence input, boolean isThrottled) {
             queryTask = null;
         }
 
-        try {
-            queryTask = new AsyncTask<URL, Void, List<CharSequence>>() {
-                @Override
-                protected List<CharSequence> doInBackground(URL... urls) {
-                    return HttpRequest.get(urls[0], userAgent);
-                }
+        queryTask = new QueryTask(view).execute(searchEngine.buildSearchSuggestionUrl(input.toString()), userAgent);
 
-                @Override
-                protected void onPostExecute(List<CharSequence> strings) {
-                    if (view != null) {
-                        view.setSuggestions(strings);
-                    }
-                }
-            }.execute(new URL(searchEngine.buildSearchSuggestionUrl(input.toString())));
-        } catch (MalformedURLException ex) {
-            // Do nothing
+
+    }
+
+    private static class QueryTask extends AsyncTask<String, Void, List<CharSequence>> {
+
+        private SoftReference<UrlInputContract.View> viewSoftReference;
+
+        QueryTask(UrlInputContract.View view) {
+            viewSoftReference = new SoftReference<>(view);
         }
 
+        @Override
+        protected List<CharSequence> doInBackground(String... strings) {
+            TrafficStats.setThreadStatsTag(SocketTags.SEARCH_SUGGESTION);
```

Comment:
Should we tag the traffic only when the strict mode is enabled?

Reply:
Is there any bad effect of tagging the traffic here? From doc I've read that tagging is a good practice for analyzing your traffics

Reply:
Just wonder if the information is needed in release build. However, it's ok for me to keep tag of traffic statistic since it should be a lightweight operation.